src/language/ast.ts

```diff
@@ -63,15 +82,97 @@ export interface ASTKindToNode {
   ModelDescription: ModelDescriptionNode;
   Name: NameNode;
   OptionalName: OptionalNameNode;
+  EndpointNamespaceTypeDefinition: EndpointNamespaceTypeDefinitionNode;
+  EndpointVerb: EndpointVerbNode;
+  EndpointUrl: EndpointUrlNode;
+  EndpointTypeDefinition: EndpointTypeDefinitionNode;
+  EndpointResponse: EndpointResponseNode;
+  SecurityDefinition: SecurityDefinitionNode;
 }
 
 export interface DocumentNode {
   readonly kind: 'Document';
   readonly loc?: Location;
-  readonly definitions: ReadonlyArray<TypeDefinitionNode>;
+  readonly definitions: ReadonlyArray<TypeDefinitionNode | EndpointNamespaceTypeDefinitionNode>;
+}
+
+export type TypeDefinitionNode =
+  | ModelTypeDefinitionNode
+  | EnumTypeDefinitionNode
+
+export interface EndpointNamespaceTypeDefinitionNode {
+  readonly kind: 'EndpointNamespaceTypeDefinition';
+  readonly tag?: NameNode;
+  readonly description?: ModelDescriptionNode;
+  readonly endpoints?: ReadonlyArray<EndpointTypeDefinitionNode>;
+  readonly loc?: Location;
+}
+
+export interface EndpointVerbNode {
+  readonly kind: 'EndpointVerb';
+  readonly name?: NameNode;
+  readonly loc?: Location;
+}
+
+
+export interface EndpointsParameterQueryNode {
+  readonly kind: 'EndpointParameterQuery';
+  readonly name?: NameNode;
+  readonly loc?: Location;
+}
+
+
+export interface EndpointParameterPathTypeNode {
+  readonly kind: 'EndpointParameterPathType',
+  readonly name: NameNode;
+  readonly loc?: Location;
 }
 
-export type TypeDefinitionNode = ModelTypeDefinitionNode | EnumTypeDefinitionNode;
+export interface EndpointParameterPathNode {
+  readonly kind: 'EndpointParameterPath';
+  readonly name?: NameNode;
+  readonly loc?: Location;
+  readonly type?: EndpointParameterPathTypeNode
+}
+
+export interface EndpointsParameterNode {
+  readonly kind: 'EndpointParameter';
+  readonly type: TypeNode | EndpointsParameterQueryNode | EndpointParameterPathNode;
```

Comment:
put in a type nearby. and why you have here TypeNode

Reply:
`TypeNode` seems like a good choice here because I used `parseTypeReference()` method for request and response (request and response) and It return TypeNode.